### ISSUE TITLE
kbpgp assumes keys with no expiry time expire in 4 years

### ISSUE THREAD

**Issue by oconnor663** [keybase/kbpgp#75]
In `keywrapper.iced`:

```coffeescript
class Lifespan
  constructor : ( {@generated, @expire_in}) ->
    @expire_in = C.default_key_expire_in unless @expire_in?
```

Is there a reason we're not respecting the does-not-expire setting of the key here?

**Comment by maxtaco** [comment]
Don't we get a 0 expire in if there is "no expire" set? In that case the
question mark operator will return true.

Ie:

0? == true

**Comment by oconnor663** [comment]
```coffeescript
iced> kbpgp = require 'kbpgp'
iced> armored = '-----BEGIN PGP PUBLIC KEY BLOCK-----\nVersion: GnuPG v2\n\nmI0EVYjdUwEEALV41J6ItewfGN/G7fXFd37qHRNtGXjKN9+YzVTDCzPV5a7KHQJi\nVlaWAnGSjVmAnQ05nin1OosMjrgTnIXFyR0O9WUmQdspAI2hn5Ph4hF6n5pMyqmg\nacUyr6f/Ar65Ht2mpfnp6+C9dM8tHWEEfU6p6XIkBNgT3675rZwHkNGBABEBAAG0\nBUphY2tPiLcEEwEIACEFAlWI3VMCGwMFCwkIBwIGFQgJCgsCBBYCAwECHgECF4AA\nCgkQCymQPWnIKrXz5QP8CUcpJRVjTvzTT/r7luDrPpn/dsU85dffa29yaBPAsYnl\n+cpl3THKEzRKLjm6vnVWzUL/g5zXG59EuEHQQa9YzUk3I6yjfDpSAj+/RnRw/z4F\ncd6CWpoJYdlwuV5QQmb5M9TOb2Ay+ZgRdyGrz0PCF/BCn1joHWqgJCy8N5jbuhW4\njQRViN1TAQQAsvCUdZYWw5c3kC2mU4JRdvqJS5INE5QYkRkdwv7GyVE8NdAeOHRq\nsTCVW1bA4aQOuGUtKpJ5jjcHAR2Tf6HKWHgoHyAe2d86Y4+nKbgEsv6CRvjDXDvb\n8fMXzTD/VCW0/yFnnzQmavzMtWQgENlg+OFw38ES6gNqt7bpkMFWfxsAEQEAAYif\nBBgBCAAJBQJViN1TAhsMAAoJEAspkD1pyCq1kY8EAI3a2OqW8pMyQEL2gWrdvBvo\nQTmwinf78f0s8F3/NAXObZkHy0VLb2ph5be34Rea38DFnZQg+CWbW7iM0UiKRHdi\nAbux5ruK4Wv6SRus/WrMnBQq2kYlshVIZR0pEJ8li/e+q3qnsO4pRl3qgPd+sBWD\nFXaICpfDSpnMOzSHp8A4\n=02c/\n-----END PGP PUBLIC KEY BLOCK-----\n'
iced> await kbpgp.ukm.import_armored_public {armored}, defer err, km
iced> km.primary.lifespan
{ generated: 1435032915, expire_in: 126144000 }
```

**Comment by maxtaco** [comment]
Ah, tis a bug!

**Comment by oconnor663** [comment]
Fixed in max's 915e53145af2d8bbf6a312339a5c5547ab7b25db and my fe6c30e5b571c4e133d23e5f48714565fb1041e1.
